godotproject/LaserBeam.gd

```diff
@@ -0,0 +1,48 @@
+extends Area2D
+
+enum ANIMATIONS {
+	IDLE, 
+	ATTACKING,
+	MAX_ANIMATIONS,
+}
+
+var AnimationStates = ["Idle", "Attacking"]
+var AnimationState = ANIMATIONS.IDLE
+var inside = false
+var player
+var attackDamage = 1
+
+# Called when the node enters the scene tree for the first time.
+func _ready():
+	$AnimatedSprite.play(AnimationStates[ANIMATIONS.IDLE])
+	print("Timer started")
+	$Timer.start()
+	
+func _process(_delta):
+	if inside == true and AnimationState == ANIMATIONS.ATTACKING:
+		player.takeDamage(attackDamage)
+		
+# If we're in an attack state do damage
+func _on_Area2D_body_entered(body):
+	inside = true
+	player = body
```

Comment:
may cause efficiency issues? I dont have any reason to back this up, just a gut feeling

Reply:
Is there a way to save a pointer to the player?

Reply:
I don't think this should cause issues, it seems like the best way to save a reference to the player